app/assets/stylesheets/components/forms.scss

```diff
@@ -0,0 +1,140 @@
+@import '../vars/_variables.scss';
+
+%form-styling {
+  background-color: $smoke-00;
+  border: 1px solid $smoke-30;
+  appearance: none;
+  border-radius: $br-default;
+  transition: all $transition-props;
+
+  &-hover {
```

Comment:
Can you tell me why you use a single dash (`-`) for `&-hover` here, when in other cases, you use a double dash (`--`), for example, in `.crayons-avatar--xl`?

Reply:
Double dash and double underscore is coming from classes naming convention of BEM approach:

- Double dash is for modifier classes,
- Double underscore is for child elements.

BEM convention, if you're not familiar with this one, is basically bunch of rules for naming classes that help make code more understandable (when you're literally reading it) AND prevents accidentally overwriting styles of other elements (because names are more specific). It's common naming convention especially when building components. More about `--` and `__` can be found [here](http://getbem.com/naming/).

---

So why this example has one dash? Because I'm actually not defining any styling there (therefor following BEM convention could be confusing). It's called "_placeholder selector_" that will then be reused in other places ([check out `%` sign couple lines above](https://github.com/thepracticaldev/dev.to/pull/6334/files/264a45b9769f97524e970808a7e91f780eacbc31#diff-0fe9bcf2fdd9908604052481e4600abcR3))

_Placeholder selector_ is more like a variable instead of real CSS class. It's just a bunch of properties stored as variable until you use it somewhere.  So here I stored `%form-styling-hover` variable which contains styling that I then use for example [here](https://github.com/thepracticaldev/dev.to/pull/6334/files/264a45b9769f97524e970808a7e91f780eacbc31#diff-0fe9bcf2fdd9908604052481e4600abcR48). It's kinda like mixins. If you're more interested in _placeholder selectors_, [there's a very good explanation](https://sass-lang.com/documentation/style-rules/placeholder-selectors).

Reply:
TIL!